drivers/network/tcpip/tcpip/iinfo.c

```diff
@@ -72,7 +72,18 @@ TDI_STATUS InfoTdiQueryGetInterfaceMIB(TDIEntityID ID,
                     case NdisPhysicalMediumUnspecified:
 #endif
                     case NdisPhysicalMediumWirelessLan:
+#if (_WIN32_WINNT >= _WIN32_WINNT_VISTA)
                         OutData->if_type = IF_TYPE_IEEE80211;
+#else
+                        /* MSDN:
+                         * IF_TYPE_IEEE80211 is returned on Windows Vista and later.
+                         * On Windows Server 2003 and Windows XP,
+                         * an IEEE 802.11 wireless network interface returns
+                         * an adapter type of MIB_IF_TYPE_ETHERNET.
+                         * https://learn.microsoft.com/en-us/windows/win32/api/iptypes/ns-iptypes-ip_adapter_info
+                         */
```

Comment:
I'm not sure a (so long) code comment is needed: the (wanted) `#if` seems enough to me.
```suggestion
```

Reply:
Indeed, and maybe copying a working from MSDN is not a good idea too. But from other side, this difference needs to be explained a bit, so we probably can use a short comment without the long text like you suggested below.

Reply:
Removed and added a short comment instead, like in other case.